lib/src/services/preferences.dart

```diff
@@ -30,6 +33,19 @@ class Preferences extends Service {
 		return result;
 	}
 
+	/// Returns the last date the data was retrieved from the
+	/// database as [DateTime].
+	DateTime? getRefreshData(String data){
```

Comment:
You say you return a `DateTime?`, but you never return null. As you saw above, that caused some confusion with the `!`. Instead of using `!` below, handle the case where you get null and spit out an error message, and that way you can return a plain `DateTime`.

Reply:
but it will be null the first time the user updates the data no?

Reply:
Yep! So the `!` will result in an error. That's why you should explicitly check for null and handle it properly. The `!` is really a last resort, similar to `dynamic` in that you're telling Dart "I know this will lead to an error if it happens, but it will never happen". You shouldn't be using it to silence the errors, because those errors usually mean you need to handle the null case.

Reply:
but I do handle it within the .getRefreshData() function

Reply:
Well firstly, it's not a good idea to say you return something you don't actually ever return -- in this case, null. It's impossible for this function to ever return null. Second, you have a ! in this function, which means you don't actually handle the null case in the HybridDataRefresh service because it hits this ! before it can be returned.